botnim/vector_store/vector_store_es.py

```diff
@@ -99,12 +99,51 @@ def _build_search_query(self, query_text: str, embedding: List[float],
         }
         
         vector_match = {
-            "knn": {
-                "field": "vector",      # field to search in
-                "query_vector": embedding,  # embedding to search for
-                "k": num_results,       # number of results to get
-                "num_candidates": 20,   # number of candidates to consider
-                "boost": 0.5           # boost for vector match
+            "bool": {
+                "should": [
+                    {
+                        "nested": {
+                            "path": "vectors",
+                            "query": {
+                                "bool": {
+                                    "must": [
+                                        {"term": {"vectors.source": "content"}},
```

Comment:
Should it be `vectors.source` or just `source` here? 
(if we're in a nested clause)

Reply:
pushed a fix, good catch

Reply:
[deleted my previous message...rechecking something]

Reply:
seems to work now.